specification/ai/OpenAI.Assistants/run_steps/models.tsp

```diff
@@ -165,3 +165,181 @@ model RunStepError {
+  type: "code_interpreter";
+
+  /** The Code Interpreter data for the tool call. */
+  @encodedName("application/json", "code_interpreter")
+  codeInterpreter?: RunStepDeltaCodeInterpreterDetailItemObject;
+}
+
+/** Represents the function data in a streaming run step delta's function tool call. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model RunStepDeltaFunction {
+  /** The name of the function. */
+  name: string;
+
+  /** The arguments passed to the function as input. */
+  arguments: string;
+
+  /** The output of the function, null if outputs have not yet been submitted. */
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "OpenAI uses explicit nullability, distinct from optionality"
+  output: string | null;
+}
+
+/** Represents the Code Interpreter tool call data in a streaming run step's tool calls. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model RunStepDeltaCodeInterpreterDetailItemObject {
+  /** The input into the Code Interpreter tool call. */
+  input: string;
+
+  /**
+   * The outputs from the Code Interpreter tool call. Code Interpreter can output one or more
+   * items, including text (`logs`) or images (`image`). Each of these are represented by a
+   * different object type.
+   */
+  outputs?: RunStepDeltaCodeInterpreterOutput[];
+}
+
+/** The abstract base representation of a streaming run step tool call's Code Interpreter tool output. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+@discriminator("type")
+model RunStepDeltaCodeInterpreterOutput {
+  /** The index of the output in the streaming run step tool call's Code Interpreter outputs array. */
+  index: int32;
+
+  /** The type of the streaming run step tool call's Code Interpreter output. */
+  type: string;
+}
+
+/** Represents a log output as produced by the Code Interpreter tool and as represented in a streaming run step's delta tool calls collection. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model RunStepDeltaCodeInterpreterLogOutput
+  extends RunStepDeltaCodeInterpreterOutput {
+  /** The type of the object, which is always "logs." */
+  type: "logs";
+
+  /** The text output from the Code Interpreter tool call. */
+  logs?: string;
+}
+
+/** Represents an image output as produced the Code interpreter tool and as represented in a streaming run step's delta tool calls collection. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model RunStepDeltaCodeInterpreterImageOutput
+  extends RunStepDeltaCodeInterpreterOutput {
+  /** The object type, which is always "image." */
+  type: "image";
+
+  /** The image data for the Code Interpreter tool call output. */
+  image?: RunStepDeltaCodeInterpreterImageOutputObject;
+}
+
+/** Represents the data for a streaming run step's Code Interpreter tool call image output. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model RunStepDeltaCodeInterpreterImageOutputObject {
+  /** The file ID for the image. */
+  @encodedName("application/json", "file_id")
```

Comment:
nit: Do we really need this? I thought the default renaming for wire format was already lower case snake. Unless, this isn't renaming the field and it's doing something else :)

Reply:
Hmm... no, we don't. I likely started this pattern back in November to ensure strict wire consistency even if we renamed (not just re-cased) the properties, but I now share the implied sentiment that this doesn't add much value (and does add spurious line count).

Either in parallel or afterwards, let's do a PR to clean it up across the board -- maybe together with comment updates per the earlier note. I'm tempted to just sneak it all in now, but that would dilute the specificity of this PR by adding in pretty much *all* the model files.